File: app/models/wkbk/folder_info.rb
# |----------+------+--------+-------------+-----------------------------|
# | 種類     | 一覧 | 直リン | 一覧条件    | 直リン表示条件              |
# |----------+------+--------+-------------+-----------------------------|
# | 公開     | ○   | ○     | public_only | true                        |
# | 限定公開 | ×   | ○     | public_only | true                        |
# | 非公開   | ×   | ×     | public_only | current_user == record.user |
# |----------+------+--------+-------------+-----------------------------|

module Wkbk
  class FolderInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: :public,  name: "公開",     show_can: -> current_user, record { true                        } },
      { key: :limited, name: "限定公開", show_can: -> current_user, record { true                        } },
      { key: :private, name: "非公開",   show_can: -> current_user, record { current_user == record.user } },
    ]
  end
end

File: front_app/components/Wkbk/models/folder_info.js
import MemoryRecord from 'js-memory-record'

export class FolderInfo extends MemoryRecord {
  static get define() {
    return [
      { key: "public",   name: "公開",     icon: "eye-outline",  type: null, message: {article: "Googleに存在がばれる",                                                           book: "トップに表示して誰でも見れる"}, },
      { key: "limited",  name: "限定公開", icon: "link",         type: null, message: {article: "リンクを知っている人だけ見れる (公開した問題集に入れるとリンクがばれるので注意)", book: "リンクを知っている人だけ見れる"}, },
      { key: "private",  name: "非公開",   icon: "lock",         type: null, message: {article: "直リンクされても見れない",                                                       book: "直リンクされても見れない"}, },
    ]
  }
}
